app/lib/efile/az/az301_calculator.rb

```diff
@@ -0,0 +1,83 @@
+module Efile
+  module Az
+    class Az301Calculator < ::Efile::TaxCalculator
+      attr_reader :lines, :value_access_tracker
+
+      def initialize(value_access_tracker:, lines:, intake:)
+        @value_access_tracker = value_access_tracker
+        @lines = lines
+        @intake = intake
+      end
+
+      def calculate
+        set_line(:AZ301_LINE_6a, method(:calculate_line_6a))
+        set_line(:AZ301_LINE_6c, method(:calculate_line_6c))
+        set_line(:AZ301_LINE_7a, method(:calculate_line_7a))
+        set_line(:AZ301_LINE_7c, method(:calculate_line_7c))
+        set_line(:AZ301_LINE_26, method(:calculate_line_26))
+        set_line(:AZ301_LINE_27, method(:calculate_line_27))
+        set_line(:AZ301_LINE_32, method(:calculate_line_32))
+        set_line(:AZ301_LINE_33, method(:calculate_line_33))
+        set_line(:AZ301_LINE_34, method(:calculate_line_34))
+        set_line(:AZ301_LINE_40, method(:calculate_line_40))
+        set_line(:AZ301_LINE_41, method(:calculate_line_41))
+        set_line(:AZ301_LINE_60, method(:calculate_line_60))
+        set_line(:AZ301_LINE_62, method(:calculate_line_62))
+      end
+
+      private
+
+      def calculate_line_6a # Credit for Contributions to Qualifying Charitable Organizations from current year
```

Comment:
ok sorry if this was covered in that CGC conversation that i got super lost in but do we not think this is redundant with `line_data.yml`?